src/Microsoft.OData.Core/ObjectPool.cs

```diff
@@ -0,0 +1,49 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ObjectPool.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.OData
+{
+    internal class ObjectPool<T>

```

Comment:
Can you please add xmldoc to the members of this class?

Reply:
Since you've made a generic object pool, I think you should also mention in the xmldoc that it's not thread-safe and not intended to be shared across concurrent requests just in case someone thinks of re-using it in different scenario.